src/labs_examples/lambdas/labs/exercise_04/Exercise_04.java

```diff
@@ -0,0 +1,155 @@
+ *      9) Demonstrate the anyMatch() function.
+ *      10) Demonstrate the allMatch() function.
+ *      11) Demonstrate the collect() terminal operation to store resulting values into a List
+ *      
+ */
+
+class Exercise_04 {
+    public static void main(String[] args) {
+        System.out.println("1) ");
+        IntStream
+                .range(1, 16)
+                .forEach(i -> System.out.print(i + " ")); // 1)
+        System.out.println();
+        System.out.println();
+
+        List<Integer> integers = Arrays.asList(1, 2, 3, 4, 33,45);  // 2)
+        Integer sum = integers.stream()
+                .mapToInt(Integer::intValue)
+                .sum();
+        System.out.println("2) The sum is: " + sum);
+        System.out.println();
+
+        List<Integer> list = Arrays.asList(3, 100, 9, 12, 15);  // 3)
+        Integer sum3 = list.stream()
+                .mapToInt(Integer::valueOf)
+                .sum();
+        System.out.println("3) =" + sum3);
+        System.out.println();
+
+
+        IntStream intStream = IntStream.of(15, 13, 45, 18, 89, 70, 76, 56); // 4)
+        OptionalDouble res = intStream
+                .filter(num -> num > 10)
+                .average();
+        System.out.print("4) Average of the elements of the stream is ");
+        if (res.isPresent()) {
+            System.out.print(res.getAsDouble());
+        } else {
+            System.out.print("Nothing!");
+        }
+        System.out.println();
+
+        List<Integer> numbers = Arrays.asList(1, 2, 3, 4, 5, 6); // 5)
+        Integer result = numbers
+                .stream()
+                .reduce(0, (element1, element2) -> element1 + element2);
+        System.out.println("5) result =  " + result);
+
+
+        String fileName = "src/labs_examples/lambdas/labs/exercise_04/line.txt"; // 6)
+        try (Stream<String> stream = Files.lines(Paths.get(fileName))) {
+            System.out.println("6) reading txt file and printing results:");
+            stream.forEach(System.out::println);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        System.out.println("end of file");
+        System.out.println();
+
+        String fileName2 = "src/labs_examples/lambdas/labs/exercise_04/stream_text_lab.csv"; // 7)
+        try (Stream<String> stream = Files.lines(Paths.get(fileName2))) {
+            System.out.println("7) reading csv file and printing results:");
+
+            File file = new File(fileName2);
+            Scanner input = new Scanner(file);
+            List<String> list5 = new ArrayList<String>();
+            while (input.hasNextLine()) {
+                list5.add(input.nextLine());
+
+            }
+            System.out.println(list5); // wasn't able to figure out how to make each element an individual string
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        System.out.println("end of file");
+        System.out.println();
```

Comment:
Streams are hard, so try this example and comment/uncomments the `peek` lines to see the output on each step. 
```suggestion
        try (Stream<String> stream = Files.lines(Paths.get(fileName2))) {
            stream
//                    .peek(s -> System.out.println("string: " + s))
                    .map(s -> s.split(","))
//                    .peek(arr -> System.out.println("arr : "+ Arrays.toString(arr)))
                    .map(arr -> arr[1])
//                    .peek(s -> System.out.println("string: " + s))
                    .forEach(System.out::println);
        } catch (IOException e) {
            e.printStackTrace();
        }
        System.out.println("end of file");
        System.out.println();
```
Let's discuss it on the meeting.